crux-core/src/crux/api/v2/CruxNode.java

```diff
@@ -0,0 +1,75 @@
+package crux.api.v2;
+
+import clojure.lang.Keyword;
+import clojure.lang.PersistentVector;
+import crux.api.Crux;
+import crux.api.ICruxAPI;
+import crux.api.IndexVersionOutOfSyncException;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static crux.api.v2.Database.database;
+import static crux.api.v2.TxResult.txResult;
+
+public class CruxNode {
+    private final ICruxAPI node;
+
+    private CruxNode(Topology topology) throws IndexVersionOutOfSyncException {
+        this.node = Crux.startNode(topology.toEdn());
+    }
+
+    public static CruxNode cruxNode(Topology topology) throws IndexVersionOutOfSyncException {
+        return new CruxNode(topology);
+    }
+
+    @SuppressWarnings("unchecked")
+    public TxResult submitTx(Iterable<Operation> ops) {
+        PersistentVector txVector = PersistentVector.create();
+        for(Operation op: ops) {
+            txVector = txVector.cons(op.toEdn());
+        }
+
+        Map<Keyword,Object> result = node.submitTx(txVector);
+        Date txTime = (Date) result.get(Keyword.intern("crux.tx/tx-time"));
```

Comment:
This is a nitpicking, but I'd replace all calls to `Keyword.intern` with private static fields for two reasons:
- make Java code a bit more understandable, showing that it is _the same_ Keyword every time
- just to save some cpu cycles on String parsing and hashmap lookup

From what I can tell, Clojure compiler does exactly this kind of per-class caching of Keywords.

Reply:
Yeah, we were just working through some of it and doing just that! :)

Reply:
Hmm, I meant just the implementation - cache the results of calls to `Keyword.intern` per class:
```
class CurxNode { 
private static final TX_TIME = Keyword.intern("crux.tx/tx-time");
...
Date txTime = (Date) result.get(TX_TIME);
}
```
Internal calls to `Keyword.intern` or possible `Util.keyword` are irrelevant for the public API

Reply:
yep, agreed :)